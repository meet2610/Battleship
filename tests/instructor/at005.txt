debug_test(medium)
fire([D,4])
bomb([C,4],[E,4])
fire([A,4])
fire([B,4])
fire([D,1])
fire([B,1])
fire([A,1])
fire([C,1])
fire([E,1])
fire([H,1])
fire([A,2])
fire([A,3])
fire([A,4])
fire([A,5])
fire([A,6])
fire([B,2])
fire([B,3])
fire([B,5])
fire([B,6])
fire([A,4])
fire([A,8])
bomb([C,1],[C,2])
bomb([C,3],[C,2])
bomb([D,3],[D,2])
bomb([E,6],[E,7])
bomb([A,7],[E,7])
bomb([E,2],[F,2])
bomb([F,4],[F,5])
--give_up
new_game(easy)
