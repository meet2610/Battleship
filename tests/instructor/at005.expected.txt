  state 0 OK -> Start a new game
->debug_test(medium)
  state 1 OK -> Fire Away!
     1  2  3  4  5  6
  A  _  _  _  _  _  _
  B  _  _  _  _  _  _
  C  _  h  h  _  _  _
  D  _  v  _  _  _  _
  E  _  v  _  _  _  _
  F  _  v  _  v  _  _
  Current Game (debug): 1
  Shots: 0/16
  Bombs: 0/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [D, 2]->v;[E, 2]->v;[F, 2]->v
    2x1: [C, 2]->h;[C, 3]->h
    1x1: [F, 4]->v
->fire([D, 4])
  state 2 OK -> Miss! Keep Firing!
     1  2  3  4  5  6
  A  _  _  _  _  _  _
  B  _  _  _  _  _  _
  C  _  h  h  _  _  _
  D  _  v  _  O  _  _
  E  _  v  _  _  _  _
  F  _  v  _  v  _  _
  Current Game (debug): 1
  Shots: 1/16
  Bombs: 0/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [D, 2]->v;[E, 2]->v;[F, 2]->v
    2x1: [C, 2]->h;[C, 3]->h
    1x1: [F, 4]->v
->bomb([C, 4],[E, 4])
  state 3 Bomb coordinates must be adjacent -> Keep Firing!
     1  2  3  4  5  6
  A  _  _  _  _  _  _
  B  _  _  _  _  _  _
  C  _  h  h  _  _  _
  D  _  v  _  O  _  _
  E  _  v  _  _  _  _
  F  _  v  _  v  _  _
  Current Game (debug): 1
  Shots: 1/16
  Bombs: 0/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [D, 2]->v;[E, 2]->v;[F, 2]->v
    2x1: [C, 2]->h;[C, 3]->h
    1x1: [F, 4]->v
->fire([A, 4])
  state 4 OK -> Miss! Keep Firing!
     1  2  3  4  5  6
  A  _  _  _  O  _  _
  B  _  _  _  _  _  _
  C  _  h  h  _  _  _
  D  _  v  _  O  _  _
  E  _  v  _  _  _  _
  F  _  v  _  v  _  _
  Current Game (debug): 1
  Shots: 2/16
  Bombs: 0/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [D, 2]->v;[E, 2]->v;[F, 2]->v
    2x1: [C, 2]->h;[C, 3]->h
    1x1: [F, 4]->v
->fire([B, 4])
  state 5 OK -> Miss! Keep Firing!
     1  2  3  4  5  6
  A  _  _  _  O  _  _
  B  _  _  _  O  _  _
  C  _  h  h  _  _  _
  D  _  v  _  O  _  _
  E  _  v  _  _  _  _
  F  _  v  _  v  _  _
  Current Game (debug): 1
  Shots: 3/16
  Bombs: 0/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [D, 2]->v;[E, 2]->v;[F, 2]->v
    2x1: [C, 2]->h;[C, 3]->h
    1x1: [F, 4]->v
->fire([D, 1])
  state 6 OK -> Miss! Keep Firing!
     1  2  3  4  5  6
  A  _  _  _  O  _  _
  B  _  _  _  O  _  _
  C  _  h  h  _  _  _
  D  O  v  _  O  _  _
  E  _  v  _  _  _  _
  F  _  v  _  v  _  _
  Current Game (debug): 1
  Shots: 4/16
  Bombs: 0/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [D, 2]->v;[E, 2]->v;[F, 2]->v
    2x1: [C, 2]->h;[C, 3]->h
    1x1: [F, 4]->v
->fire([B, 1])
  state 7 OK -> Miss! Keep Firing!
     1  2  3  4  5  6
  A  _  _  _  O  _  _
  B  O  _  _  O  _  _
  C  _  h  h  _  _  _
  D  O  v  _  O  _  _
  E  _  v  _  _  _  _
  F  _  v  _  v  _  _
  Current Game (debug): 1
  Shots: 5/16
  Bombs: 0/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [D, 2]->v;[E, 2]->v;[F, 2]->v
    2x1: [C, 2]->h;[C, 3]->h
    1x1: [F, 4]->v
->fire([A, 1])
  state 8 OK -> Miss! Keep Firing!
     1  2  3  4  5  6
  A  O  _  _  O  _  _
  B  O  _  _  O  _  _
  C  _  h  h  _  _  _
  D  O  v  _  O  _  _
  E  _  v  _  _  _  _
  F  _  v  _  v  _  _
  Current Game (debug): 1
  Shots: 6/16
  Bombs: 0/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [D, 2]->v;[E, 2]->v;[F, 2]->v
    2x1: [C, 2]->h;[C, 3]->h
    1x1: [F, 4]->v
->fire([C, 1])
  state 9 OK -> Miss! Keep Firing!
     1  2  3  4  5  6
  A  O  _  _  O  _  _
  B  O  _  _  O  _  _
  C  O  h  h  _  _  _
  D  O  v  _  O  _  _
  E  _  v  _  _  _  _
  F  _  v  _  v  _  _
  Current Game (debug): 1
  Shots: 7/16
  Bombs: 0/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [D, 2]->v;[E, 2]->v;[F, 2]->v
    2x1: [C, 2]->h;[C, 3]->h
    1x1: [F, 4]->v
->fire([E, 1])
  state 10 OK -> Miss! Keep Firing!
     1  2  3  4  5  6
  A  O  _  _  O  _  _
  B  O  _  _  O  _  _
  C  O  h  h  _  _  _
  D  O  v  _  O  _  _
  E  O  v  _  _  _  _
  F  _  v  _  v  _  _
  Current Game (debug): 1
  Shots: 8/16
  Bombs: 0/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [D, 2]->v;[E, 2]->v;[F, 2]->v
    2x1: [C, 2]->h;[C, 3]->h
    1x1: [F, 4]->v
->fire([H, 1])
  state 11 Invalid coordinate -> Keep Firing!
     1  2  3  4  5  6
  A  O  _  _  O  _  _
  B  O  _  _  O  _  _
  C  O  h  h  _  _  _
  D  O  v  _  O  _  _
  E  O  v  _  _  _  _
  F  _  v  _  v  _  _
  Current Game (debug): 1
  Shots: 8/16
  Bombs: 0/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [D, 2]->v;[E, 2]->v;[F, 2]->v
    2x1: [C, 2]->h;[C, 3]->h
    1x1: [F, 4]->v
->fire([A, 2])
  state 12 OK -> Miss! Keep Firing!
     1  2  3  4  5  6
  A  O  O  _  O  _  _
  B  O  _  _  O  _  _
  C  O  h  h  _  _  _
  D  O  v  _  O  _  _
  E  O  v  _  _  _  _
  F  _  v  _  v  _  _
  Current Game (debug): 1
  Shots: 9/16
  Bombs: 0/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [D, 2]->v;[E, 2]->v;[F, 2]->v
    2x1: [C, 2]->h;[C, 3]->h
    1x1: [F, 4]->v
->fire([A, 3])
  state 13 OK -> Miss! Keep Firing!
     1  2  3  4  5  6
  A  O  O  O  O  _  _
  B  O  _  _  O  _  _
  C  O  h  h  _  _  _
  D  O  v  _  O  _  _
  E  O  v  _  _  _  _
  F  _  v  _  v  _  _
  Current Game (debug): 1
  Shots: 10/16
  Bombs: 0/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [D, 2]->v;[E, 2]->v;[F, 2]->v
    2x1: [C, 2]->h;[C, 3]->h
    1x1: [F, 4]->v
->fire([A, 4])
  state 14 Already fired there -> Keep Firing!
     1  2  3  4  5  6
  A  O  O  O  O  _  _
  B  O  _  _  O  _  _
  C  O  h  h  _  _  _
  D  O  v  _  O  _  _
  E  O  v  _  _  _  _
  F  _  v  _  v  _  _
  Current Game (debug): 1
  Shots: 10/16
  Bombs: 0/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [D, 2]->v;[E, 2]->v;[F, 2]->v
    2x1: [C, 2]->h;[C, 3]->h
    1x1: [F, 4]->v
->fire([A, 5])
  state 15 OK -> Miss! Keep Firing!
     1  2  3  4  5  6
  A  O  O  O  O  O  _
  B  O  _  _  O  _  _
  C  O  h  h  _  _  _
  D  O  v  _  O  _  _
  E  O  v  _  _  _  _
  F  _  v  _  v  _  _
  Current Game (debug): 1
  Shots: 11/16
  Bombs: 0/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [D, 2]->v;[E, 2]->v;[F, 2]->v
    2x1: [C, 2]->h;[C, 3]->h
    1x1: [F, 4]->v
->fire([A, 6])
  state 16 OK -> Miss! Keep Firing!
     1  2  3  4  5  6
  A  O  O  O  O  O  O
  B  O  _  _  O  _  _
  C  O  h  h  _  _  _
  D  O  v  _  O  _  _
  E  O  v  _  _  _  _
  F  _  v  _  v  _  _
  Current Game (debug): 1
  Shots: 12/16
  Bombs: 0/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [D, 2]->v;[E, 2]->v;[F, 2]->v
    2x1: [C, 2]->h;[C, 3]->h
    1x1: [F, 4]->v
->fire([B, 2])
  state 17 OK -> Miss! Keep Firing!
     1  2  3  4  5  6
  A  O  O  O  O  O  O
  B  O  O  _  O  _  _
  C  O  h  h  _  _  _
  D  O  v  _  O  _  _
  E  O  v  _  _  _  _
  F  _  v  _  v  _  _
  Current Game (debug): 1
  Shots: 13/16
  Bombs: 0/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [D, 2]->v;[E, 2]->v;[F, 2]->v
    2x1: [C, 2]->h;[C, 3]->h
    1x1: [F, 4]->v
->fire([B, 3])
  state 18 OK -> Miss! Keep Firing!
     1  2  3  4  5  6
  A  O  O  O  O  O  O
  B  O  O  O  O  _  _
  C  O  h  h  _  _  _
  D  O  v  _  O  _  _
  E  O  v  _  _  _  _
  F  _  v  _  v  _  _
  Current Game (debug): 1
  Shots: 14/16
  Bombs: 0/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [D, 2]->v;[E, 2]->v;[F, 2]->v
    2x1: [C, 2]->h;[C, 3]->h
    1x1: [F, 4]->v
->fire([B, 5])
  state 19 OK -> Miss! Keep Firing!
     1  2  3  4  5  6
  A  O  O  O  O  O  O
  B  O  O  O  O  O  _
  C  O  h  h  _  _  _
  D  O  v  _  O  _  _
  E  O  v  _  _  _  _
  F  _  v  _  v  _  _
  Current Game (debug): 1
  Shots: 15/16
  Bombs: 0/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [D, 2]->v;[E, 2]->v;[F, 2]->v
    2x1: [C, 2]->h;[C, 3]->h
    1x1: [F, 4]->v
->fire([B, 6])
  state 20 OK -> Miss! Keep Firing!
     1  2  3  4  5  6
  A  O  O  O  O  O  O
  B  O  O  O  O  O  O
  C  O  h  h  _  _  _
  D  O  v  _  O  _  _
  E  O  v  _  _  _  _
  F  _  v  _  v  _  _
  Current Game (debug): 1
  Shots: 16/16
  Bombs: 0/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [D, 2]->v;[E, 2]->v;[F, 2]->v
    2x1: [C, 2]->h;[C, 3]->h
    1x1: [F, 4]->v
->fire([A, 4])
  state 21 No shots remaining -> Keep Firing!
     1  2  3  4  5  6
  A  O  O  O  O  O  O
  B  O  O  O  O  O  O
  C  O  h  h  _  _  _
  D  O  v  _  O  _  _
  E  O  v  _  _  _  _
  F  _  v  _  v  _  _
  Current Game (debug): 1
  Shots: 16/16
  Bombs: 0/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [D, 2]->v;[E, 2]->v;[F, 2]->v
    2x1: [C, 2]->h;[C, 3]->h
    1x1: [F, 4]->v
->fire([A, 8])
  state 22 No shots remaining -> Keep Firing!
     1  2  3  4  5  6
  A  O  O  O  O  O  O
  B  O  O  O  O  O  O
  C  O  h  h  _  _  _
  D  O  v  _  O  _  _
  E  O  v  _  _  _  _
  F  _  v  _  v  _  _
  Current Game (debug): 1
  Shots: 16/16
  Bombs: 0/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [D, 2]->v;[E, 2]->v;[F, 2]->v
    2x1: [C, 2]->h;[C, 3]->h
    1x1: [F, 4]->v
->bomb([C, 1],[C, 2])
  state 23 Already fired there -> Keep Firing!
     1  2  3  4  5  6
  A  O  O  O  O  O  O
  B  O  O  O  O  O  O
  C  O  h  h  _  _  _
  D  O  v  _  O  _  _
  E  O  v  _  _  _  _
  F  _  v  _  v  _  _
  Current Game (debug): 1
  Shots: 16/16
  Bombs: 0/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [D, 2]->v;[E, 2]->v;[F, 2]->v
    2x1: [C, 2]->h;[C, 3]->h
    1x1: [F, 4]->v
->bomb([C, 3],[C, 2])
  state 24 OK -> 2x1 ship sunk! Keep Firing!
     1  2  3  4  5  6
  A  O  O  O  O  O  O
  B  O  O  O  O  O  O
  C  O  X  X  _  _  _
  D  O  v  _  O  _  _
  E  O  v  _  _  _  _
  F  _  v  _  v  _  _
  Current Game (debug): 1
  Shots: 16/16
  Bombs: 1/3
  Score: 2/6 (Total: 2/6)
  Ships: 1/3
    3x1: [D, 2]->v;[E, 2]->v;[F, 2]->v
    2x1: [C, 2]->X;[C, 3]->X
    1x1: [F, 4]->v
->bomb([D, 3],[D, 2])
  state 25 OK -> Hit! Keep Firing!
     1  2  3  4  5  6
  A  O  O  O  O  O  O
  B  O  O  O  O  O  O
  C  O  X  X  _  _  _
  D  O  X  O  O  _  _
  E  O  v  _  _  _  _
  F  _  v  _  v  _  _
  Current Game (debug): 1
  Shots: 16/16
  Bombs: 2/3
  Score: 2/6 (Total: 2/6)
  Ships: 1/3
    3x1: [D, 2]->X;[E, 2]->v;[F, 2]->v
    2x1: [C, 2]->X;[C, 3]->X
    1x1: [F, 4]->v
->bomb([E, 6],[E, 7])
  state 26 Invalid coordinate -> Keep Firing!
     1  2  3  4  5  6
  A  O  O  O  O  O  O
  B  O  O  O  O  O  O
  C  O  X  X  _  _  _
  D  O  X  O  O  _  _
  E  O  v  _  _  _  _
  F  _  v  _  v  _  _
  Current Game (debug): 1
  Shots: 16/16
  Bombs: 2/3
  Score: 2/6 (Total: 2/6)
  Ships: 1/3
    3x1: [D, 2]->X;[E, 2]->v;[F, 2]->v
    2x1: [C, 2]->X;[C, 3]->X
    1x1: [F, 4]->v
->bomb([A, 7],[E, 7])
  state 27 Bomb coordinates must be adjacent -> Keep Firing!
     1  2  3  4  5  6
  A  O  O  O  O  O  O
  B  O  O  O  O  O  O
  C  O  X  X  _  _  _
  D  O  X  O  O  _  _
  E  O  v  _  _  _  _
  F  _  v  _  v  _  _
  Current Game (debug): 1
  Shots: 16/16
  Bombs: 2/3
  Score: 2/6 (Total: 2/6)
  Ships: 1/3
    3x1: [D, 2]->X;[E, 2]->v;[F, 2]->v
    2x1: [C, 2]->X;[C, 3]->X
    1x1: [F, 4]->v
->bomb([E, 2],[F, 2])
  state 28 OK -> 3x1 ship sunk! Game Over!
     1  2  3  4  5  6
  A  O  O  O  O  O  O
  B  O  O  O  O  O  O
  C  O  X  X  _  _  _
  D  O  X  O  O  _  _
  E  O  X  _  _  _  _
  F  _  X  _  v  _  _
  Current Game (debug): 1
  Shots: 16/16
  Bombs: 3/3
  Score: 5/6 (Total: 5/6)
  Ships: 2/3
    3x1: [D, 2]->X;[E, 2]->X;[F, 2]->X
    2x1: [C, 2]->X;[C, 3]->X
    1x1: [F, 4]->v
->bomb([F, 4],[F, 5])
  state 29 Game not started -> Start a new game
     1  2  3  4  5  6
  A  O  O  O  O  O  O
  B  O  O  O  O  O  O
  C  O  X  X  _  _  _
  D  O  X  O  O  _  _
  E  O  X  _  _  _  _
  F  _  X  _  v  _  _
  Current Game (debug): 1
  Shots: 16/16
  Bombs: 3/3
  Score: 5/6 (Total: 5/6)
  Ships: 2/3
    3x1: [D, 2]->X;[E, 2]->X;[F, 2]->X
    2x1: [C, 2]->X;[C, 3]->X
    1x1: [F, 4]->v
->new_game(easy)
  state 30 OK -> Fire Away!
     1  2  3  4
  A  _  _  _  _
  B  _  _  _  _
  C  _  _  _  _
  D  _  _  _  _
  Current Game: 1
  Shots: 0/8
  Bombs: 0/2
  Score: 0/3 (Total: 0/3)
  Ships: 0/2
    2x1: Not Sunk
    1x1: Not Sunk
